### PR TITLE
Fueltank Balancing

### PR DESCRIPTION
**Pull request by ZomgPonies**
Potential breaches on heavy ranges are important to give a more random breach pattern. However, we need the ability to destroy walls (using 'heavy' damage) while being guaranteed not to breach floors.

The idea here is that if someone sets and explosion with a devastation range of 0, the chances are that they don't want any breaches at all in his explosion radius. This passes the devastation range to turf/simulated/floors/ex_act() so it can react accordingly.

- Fueltanks are easy to blow up, either by mistake (refueling) or intentionally (one shot).
- Fueltanks currently lead to breaches.
- Breaches are pretty deadly.
- Breaches lead to boinks.
- Fueltanks have all sorts of underutilized gameplay functions (they are assemblies, so all the fun remote/infrared/voice activation/proximity shenanigans).
- Let's minimize boinks that could be better handled ICly.

### CODE REVIEW

**Comment by jughu** [comment]
yes but that brings the factor of a dangerous envoirement if you shoot a tank you cause a breach but then again beware of you surroundings and dont shoot randomly.

and accidently bombing yourself with a welding tool, what can i say accidents happen just make it require harm intent or something!

**Comment by KasparoVy** [comment]
RIP tactical fuel tanks. I've put them to damn good use up until now.

**Comment by deep-fried-faceplant** [comment]
I understand wanting to make fuel tanks less deadly so that minor accidents aren't punished as harshly. However, "Let's minimize boinks that could be better handled ICly" implies to me that fueltank bombing is being pushed from an OOC rule issue to a SL issue. By removing the ability for fuel tank explosions to cause floor breaches, using them to bomb a department is now significantly safer and with less collateral damage. Greytiders might see this as a reason to bomb departments, if the fuel tank explosion is away from external walls and windows. Currently, a fuel tank explosion in front of botany would definitely be worthy of a bwoink, because it would breach a major hallway. With this change, the explosion would still cause some damage to the front of botany and anyone too close to the explosion, but the damage would be localized at botany, rather than being spread out across the entire breached hallway. There are some hypothetical situations where players, as non-antagonists and still roleplaying their characters, would definitely use a fuel tank bomb, if the damage was nerfed. The memory of the riots Para-sec endured months ago are still fresh with me, and while efforts were made to squash that terrible meme, I feel as though it wouldn't take much to rile up the current crowd to start a similar situation. Packs of greytiders, cliques, vox kin-ships, and just plain stupid lone incidents are all possible sources of this hypothetical situation, of someone fuel tank bombing a location to escalate some conflict, with the reasoning that fuel tank explosions are not as deadly as before.

Maybe I'm just reading too much into it, but this is a grey (hah) zone to consider when making this change. Players will do what the code base lets them, and while this change might reduce admin intervention in accidental and innocent situations, it could increase the amount of malicious fuel tank bombings, which would need to be addressed either through bwoinks or batons.

To put it in a simplier quote, lets give it a shot, +1.

**Comment by Shadeykins** [comment]
If I'm understanding this PR correctly, fueltanks can now be used to blow up nearby walls without breaching the floor anymore - which is a great change. It adds much more utility to fuel tanks without having to worry about accidentally murdering yourself via a floor breach.

As a digression, explosions don't seem to reliably destroy girders on walls (they like to strip the metal plating off, but not remove the girder). This PR obviously doesn't touch on how explosives handle, but it would be nice to see explosions more reliably reduce walls to floor plating (C4 and X4 have a habit of leaving a girder behind, making them unreliable and leaving an unpassable girder despite "blowing the wall out").

Either way, +1

**Comment by Vivalas** [comment]
Eh, this seems like a closet hugbox masquerading as a way to make fueltanks more useful.

Assemblies are really cool with fuel tanks, but I feel removing the ability to use fuel tanks to cause serious damage will outweight a single niche use that will now be possible.

**Comment by ZomgPonies** [comment]
I'd argue that fueltanks were never meant to be this deadly in the first place. They'd been severely nerfed already in the past, and a 1 tile breach was a pretty minor fix before fastmos. This is simply an adjustment for the way fastmos acts, making even a 1 tile breach a deadly event, especially with the way spacewind blasts you around in enclosed spaces.

As for greytiders using them, that would fall under the same rules as any other random griefing. Quite frankly even with greytiders (not actual griefers) using them, as long as noone gets hurt I believe it's the kind of thing that should be handled ICly by spacelaw, not through admin babysitting.

**Comment by ZomgPonies** [comment]
@Vivalas I'd argue that the fact that they do so much damage for so little effort puts them already in a single-user niche: lazy antags who won't get boinked by admins.

**Comment by Vivalas** [comment]
Perhaps, but I rarely ever see fuel tanks used in a useful matter by antags, and it's almost never creative. 

My favorite use of fuel tanks as security is shooting antags dragging them behind them, because it almost always only fucks them over. However, they can be used in much more intelligent ways that this would hinder, such as rigging them with proximity sensors in dark corners of maintenance. When security runs buy on the chase, KABOOM, no more officer.

Fastmos brought back some danger to the game, which is great, and we don't need to retroactively nerf things that were unopposed before just to try to go back to the status quo.

File: code/game/objects/explosion.dm [hunk 1]
```diff
@@ -138,7 +138,7 @@
 							if(AM && AM.simulated)
 								AM.ex_act(dist)
 							CHECK_TICK
-					T.ex_act(dist)
+					T.ex_act(dist, devastation_range)
```
Comment (Fox-McCloud) [inline code/game/objects/explosion.dm]: There's no arguments for this anywhere.

**Comment by Fox-McCloud** [comment]
This isn't just "Fuel tank balancing"; this literally nerfs and alters every explosion in the game.

File: code/game/turfs/simulated/floor.dm [hunk 2]
```diff
@@ -51,27 +51,35 @@ var/list/icons_to_ignore_at_floor_init = list("damaged1","damaged2","damaged3","
...
-				if(3)
-					if(prob(80))
-						src.break_tile_to_plating()
-					else
-						src.break_tile()
-					src.hotspot_expose(1000,CELL_VOLUME)
-					if(prob(33)) new /obj/item/stack/sheet/metal(src)
+			else
+				if(prob(80))
+					src.break_tile_to_plating()
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/floor.dm]: if you're going to add in new code, don't add in implied `src`

File: code/game/turfs/simulated/floor.dm [hunk 3]
```diff
@@ -51,27 +51,35 @@ var/list/icons_to_ignore_at_floor_init = list("damaged1","damaged2","damaged3","
...
+				else
+					src.break_tile()
+				src.hotspot_expose(1000,CELL_VOLUME)
+				if(prob(33)) new /obj/item/stack/sheet/metal(src)
```
Comment (Fox-McCloud) [inline code/game/turfs/simulated/floor.dm]: New line.

**Comment by Tayyyyyyy** [comment]
Now that Fox has pointed out that this nerfs every explosion in the game, maybe this is too heavy handed? Explosions are supposed to be deadly in space and cause severe issues cause there's only a layer of metal separating you from the void.

I do like Shadeykin's idea of an explosive that is good for just breaching walls/doors though.

**Comment by deep-fried-faceplant** [comment]
Thought about it again, and I'm changing my +1 to a -1. On one hand, I would like to see what's covered by SL to be expanded and to generally get admin intervention out of IC issues, but fuel-tank bombs are still fairly damaging, even without breaches. If the precedent gets set that using them to bomb departments under certain circumstances is okay, then like with all rules, the border that people will infringe upon will be pushed a little further. While being able to fuel tank bomb a department for a riot does sound cool, the overall increase of chaos, and by extension the increased workload for both security and admins, does not sound like something that Paradise needs more of.

**Comment by TullyBurnalot** [comment]
A change to how fuel tanks work so that they work more like an improvised tactical breaching option and less of a "Woops, welder was on, hallway's depressurized now" would be interesting. However, the way this is handled right now affects EVERY explosion the game will produce, which is rather far-ranging for what's meant to be a change to fuel tanks. We'd appreciate it if you could keep the changes to only what this is meant to affect.

**Comment by tigercat2000** [comment]
This needs a response or it'll be closed.

**Comment by Vivalas** [comment]
I think "appreciate" is a bad way to put it... atomizing your changes is like, a basic rule of contributing to any open source codebase, as well as not changing the behavior of every explosion in the game without mentioning it in the PR.

**Comment by Aurorablade** [comment]
@ZomgPonies look here.

**Comment by tigercat2000** [comment]
No updates in 17 days. Feel free to reopen this PR with the requested changes.

**Pull request closed without merging**